tasks/read-a-configuration-file/src/main.rs

```diff
@@ -28,12 +28,51 @@ impl ConfigVariable {
     }
 }
 
+trait FromConfig {
+    fn from_config(param: &ConfigParams, key: &str) -> Result<Self, String>
+        where Self: std::marker::Sized;
```

Comment:
Note that you don't need to specify the whole path of the `Sized` trait. Writing `where Self: Sized` would be enough.

Additionally, I have some comments regarding style, but feel free to ignore them in case you don't agree. In your code you disallow the trait from being implemented by types that are `?Sized`, like `str` or `[T]`. While the current implementation certainly achieves this goal, the intent of the code may not be clear at a first sight. A different way to achieve the same result would be to define the trait as `trait FromConfig: Sized`. IMO this would make things clearer, but in the end it is just a matter of style, so feel free to ignore this.